code/game/machinery/newscaster.dm

```diff
@@ -973,6 +974,13 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 		else if(href_list["refresh"])
 			updateUsrDialog()
 
+/obj/machinery/newscaster/proc/assassination_check(var/obj/item/weapon/photo/P)
+	if (assassination_objectives.len > 0)
+		for (var/datum/objective/target/assassinate/ass in assassination_objectives)
+			for(var/ass_id in P.double_agent_completion_ids)
+				if (ass.double_agent_completion_id == ass_id)
```

Comment:
```suggestion
			if(ass.double_agent_completion_id in P.double_agent_completion_ids)
```
Much better if you would drop this strange snowflaked UId system and used weakrefs instead.